apps/dapp-console/app/settings/components/ClaimRebateDialog.tsx

```diff
@@ -8,39 +8,50 @@ import {
   DialogTrigger,
 } from '@eth-optimism/ui-components'
 import { Text } from '@eth-optimism/ui-components/src/components/ui/text/text'
-import { Address, Transaction, formatEther } from 'viem'
+import { formatEther } from 'viem'
 import { useCallback, useMemo, useState } from 'react'
 import { MAX_CLAIMABLE_AMOUNT } from '@/app/constants/rebate'
 import { shortenAddress } from '@eth-optimism/op-app'
 import Link from 'next/link'
+import { Contract } from '@/app/types/api'
+import { Network } from '@/app/components/Network'
+import { optimism } from 'viem/chains'
+import { apiClient } from '@/app/helpers/apiClient'
 
 export type ClaimRebateDialogProps = {
-  contractAddress: Address
-  contractDeploymentTransaction: Transaction
-  amountClaimed?: bigint
-  isVerified?: boolean
+  contract: Contract
   children: React.ReactNode
 }
 
 export const ClaimRebateDialog = ({
-  contractAddress,
-  contractDeploymentTransaction,
-  amountClaimed = BigInt(0),
-  isVerified = false,
+  contract,
   children,
 }: ClaimRebateDialogProps) => {
   const [isOpen, setOpen] = useState(false)
 
+  const { data: wallets } = apiClient.wallets.listWallets.useQuery({})
+  const firstVerifiedWallet = useMemo(
+    () => wallets?.records.find((wallet) => wallet.verifications.isCbVerified),
+    [wallets],
+  )
+
+  const { data: totalRebateAmount } =
+    apiClient.Rebates.totalRebatesClaimed.useQuery()
+  const amountClaimed = useMemo(
+    () => totalRebateAmount ?? BigInt(0),
+    [totalRebateAmount],
+  )
+
   const gasFeeToReimburst = useMemo(() => {
-    const gasAmount = contractDeploymentTransaction.gas
+    const gasAmount = BigInt(contract.transaction?.gasUsed ?? 0)
```

Comment:
should this be gasUsed * gasPrice? that is how we calculate it on the backend: https://github.com/ethereum-optimism/ecosystem/blob/dfaba331691e610cdcbb4154b43778007709e17e/apps/dapp-console-api/src/routes/rebates/RebatesRoute.ts#L261